README.md

```diff
@@ -5,23 +5,52 @@ Overview
 
 An instance of Transformer is constructed with a Config object.
 
-Each config object specifies a constructor function (eg., a Tyranid
-collection constructor), and an array of FieldConfig objects.
+Each Config object specifies an array of FieldConfig objects, along
+with an optional constructor function (eg., a Tyranid
+collection constructor).
+
+If a constructor function is given, it is invoked with the result
+object.
+Otherwise, the result object is directly returned.
 
 Each FieldConfig object identifies a field on the target object
 (ie., a Tyranid document), and a selector that can either be a field
-name, or a path into the source object, represented by a list of field names.
+name, or a array of names that represents a path into the source
+object.
+As a convenient shorthand, if a selector isn't given for a particular
+instance of FieldConfig, its name field is used as a selector.
```

Comment:
Non-blocking/future improvement: Would be nice to have a small code snippet accompanying each description for developers to reference.